Update why with clearer intro and dark mode images

mmccoyd opened the pull request:
Split this into an intro of reasons of: size, cool and layers. After that comes why layers help. Plus much nicer dark mode images.

mmccoyd committed:
Why: Put 40% benefit and layer diagram earlier.

- Full size, Half require a move
- Split layer access to para after layer diagram
- Split sentence
- Start 40 para with their result, reduced stretching
- Some like small with cool keycaps, cases. Big is just bulky
- Verbs instead of nouns
- Just tablet, skip phone
- "to be under"

mmccoyd committed:
Why: Tighten explanations, dark mode images

- Tighten full-size use of shift layer for capitals example
- Starter 12-13x4 gives letters and space/mods/layer keys with low movement and simple layering
- Tighten num row is sometimes handy
- Tighten beginner example layout
- Tradeoffs list

Images:
- Dark mode
- Capitals layer in mini keymap example
- Sys, not mods, on most tucked thumb/middle column

mmccoyd committed:
Why: images: center mods not duplicated

- Esc to stronger finger
- No mod duplication in center
- List OSM on layered home-row mods
- Show other keys on pressed thumb arc as inaccessible
- Example layers: show two edit keys

mmccoyd committed:
Why: intro summarizing reasons, cleanup layers

- intro reasons of size, cool, layers
- compare to laptop board, not fullsize
- remove 'sometimes ignoring ergonomics'
- cleanup board size and example sections

mmccoyd committed:
Why: tighter intro section on layering

mmccoyd committed:
Why: split lines at phrase breaks for easier diffs

mmccoyd committed:
Why: cleanup example, remove extra 12u start bits

mmccoyd committed:
Why: split layout into own para

- Good layering moves keys closer

mmccoyd committed:
Why: clarify types of board use

mmccoyd committed:
Why: change end trade-off section to just mention

mmccoyd committed:
Why: tighten intro; mention far arrows and delete

mmccoyd committed:
Why: tighten grammer

mmccoyd committed:
Why: cleanup wording, put final ref above image

FrancisUsher commented on image/hillside_keymap.png:
- The contrast on the original image seems a lot better. Can you use colors with a lower Lightness value in the dark-mode version?
- The purple-ish default background color seems unnecessary, and doesn't really play well with the other colors. Why not go with a more neutral, less Saturated color closer to a dark gray?

Even though I'm a heavy dark mode user I think the original image looks easier to read

mmccoyd replied:
Gray would be better. 

I'd used draw.io and its color pallets, but one of the new(ish) keymap drawing tools could give more control. Was that your thought below?

FrancisUsher replied:
I was just thinking more muted, darker colors would look better in the context of a dark mode screen. And the light text would show up better.

FrancisUsher commented on image/layers_snip.drawio.png:
Same comment applies here about enhancing the contrast and odd choice of default background color. Is this Handarbeit? 😝

mmccoyd replied:
Does that mean "by hand"? I guess roughly yes.

FrancisUsher replied:
Aha no I meant GMK Handarbeit, a classic keycap set with a famously loud[ and dissonant color palette](https://novelkeys.com/cdn/shop/products/gmk_handarbeit_nachtarbeit_kit_01_handarbeit_b_1200x.png?v=1689883207) . Just a joke, it's not quite that bad

FrancisUsher commented on why.md:
```diff
@@ -8,24 +8,110 @@ editor: markdown
-# Why use a 40% keyboard?
 
-On a full-size keyboard, pressing less frequent keys involves moving your hands or stretching your fingers to hit small distant targets and then reorienting to the home row. That reaching is slow and can lead to hand issues; a full-size keyboard also uses more desk space and is hard to throw in a bag for tablet use in a cafe, classroom or plane.
+Although people have different reasons for using 40% keyboards, some common reasons are:
```
This line seems obvious and unnecessary. If the title is "Why use a 40%?" you can just start listing reasons right away.

mmccoyd replied:
Partly to say "Why you like them isn't wrong, but we'll just talk about these three. " And as a more gradual intro to the piece. 

That has been reworked many times. "While people..." was another option. I'll think on it more.

FrancisUsher commented on why.md:
```diff
@@ -8,24 +8,110 @@ editor: markdown
+Although people have different reasons for using 40% keyboards, some common reasons are:
 
-A 40% keyboard uses layer-shift keys to move less frequent keys into layers under the fingers. Pressing a combination of easy-to-reach keys can be faster and easier than reaching for a distant key. Full-size keyboards already use shift keys for capital letters and most symbols; we don't have a larger keyboard with separate keys for each. 40% keyboards apply this same idea to a smaller set of base ...
+- Small boards are easy to carry and fit on small surfaces.
+- Small is cool, cute or elegant, based on taste.
+- Good layers move many keys 'close', making small more efficient than large.
```
I like these short, bite-sized, easy-to-digest bullet points!

FrancisUsher commented:
@mmccoyd I'm curious what's the reason for splitting some sentences onto multiple lines in the .MD source? I'm assuming those line breaks are not intended to render in the output content. Is there a rule of thumb you used for where line-breaks should occur in mid-sentence? If so, maybe it would be worth adding some sort of style guide for contributors?

It looks very poetic :)

FrancisUsher reviewed (commented):
@mmccoyd I'm curious what's the reason for splitting some sentences onto multiple lines in the .MD source? I'm assuming those line breaks are not intended to render in the output content. Is there a rule of thumb you used for where line-breaks should occur in mid-sentence? If so, maybe it would be worth adding some sort of style guide for contributors? It looks very poetic :)

Looks pretty good overall, I like how you tightened up a lot of the text. Especially on the pages like the "why" page, it makes sense to keep the points bite-sized wherever possible.

mmccoyd commented:
Thanks for the comments! I'll take another pass on the words and see about a better graphic tool.

The line breaks are habit from writing things collaboratively (in Latex) within git repos where a diff on a 12 word line is more workable than a diff on a 200 word line... With the breaks falling on clause breaks within the sentences. I recently learned it is called semantic line breaks. (https://sembr.org/). I'd written the first version in a more normal paragraph wrapping editor, so it may be a bit inconsistent.

FrancisUsher commented:
> The line breaks are habit from writing things collaboratively (in Latex) within git repos where a diff on a 12 word line is more workable than a diff on a 200 word line... With the breaks falling on clause breaks within the sentences. I recently learned it is called semantic line breaks. (https://sembr.org/). I'd written the first version in a more normal paragraph wrapping editor, so it may be a bit inconsistent.

Cool! I love that kind of convention, it looks very practical. Would definitely suggest adding a comment at least recommending e.g. sembr to the README or something like that, so people know where to go to read about it and can try using it if they feel comfortable.

mmccoyd committed:
Why: dark mode images more muted, better contrast

mmccoyd committed:
Why: tweak intro and heading levels

mmccoyd committed:
Why: strip trailing white space

mmccoyd commented:
Updated the images to more muted colors but with good font to background color contrast, tweaked the verbiage a tiny bit and added an image_readme file.

FrancisUsher commented:
Really great fixes. The key map looks way better now.

FrancisUsher reviewed (approved):
Reviewed and approved, nice colors and improved contrast especially.

mmccoyd committed:
Why: takes up less desk space

mmccoyd committed:
Why: need less desk space

mmccoyd committed:
Why: tighter first sentence, clearer words grammar

mmccoyd committed:
Why: intro: "People have different keyboard needs"
